gms/endpoint_state.cc

```diff
@@ -61,6 +62,33 @@ future<> i_endpoint_state_change_subscriber::on_application_state_change(inet_ad
     return make_ready_future<>();
 }
 
+locator::host_id endpoint_state::get_host_id() const {
+    if (auto* app_state = get_application_state_ptr(application_state::HOST_ID)) {
+        return locator::host_id(utils::UUID(app_state->value()));
+    }
+    return locator::host_id::create_null_id();
+}
+
+std::optional<locator::endpoint_dc_rack> endpoint_state::get_dc_rack() const {
+    if (auto* app_state = get_application_state_ptr(application_state::DC)) {
+        std::optional<locator::endpoint_dc_rack> ret;
+        ret->dc = app_state->value();
+        if ((app_state = get_application_state_ptr(application_state::RACK))) {
+            ret->rack = app_state->value();
+            return ret;
+        }
+    }
+    return std::nullopt;
+}
+
+std::unordered_set<dht::token> endpoint_state::get_tokens() const {
+    std::unordered_set<dht::token> ret;
+    if (auto* app_state = get_application_state_ptr(application_state::TOKENS)) {
+        ret = versioned_value::tokens_from_string(app_state->value());
+    }
+    return ret;
```

Comment:
Should we return optional here too? so before using it, you must check if it's engaged.